File: csharp/eks-cluster-with-prometheus-grafana/src/EksClusterWithPrometheusGrafana/VPCStack.cs
using System;
using System.Collections.Generic;
using System.Text;
using Amazon.CDK;
using Amazon.CDK.AWS.EC2;

namespace EksClusterWithPrometheusGrafana
{
    public class VPCStack : Stack
    {
        public string VPCID { get; private set; }
        public Vpc VPC { get; private set; }

        internal VPCStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            var vpc = new Vpc(this, Constants.VPC_ID, new VpcProps
            {
                Cidr = Vpc.DEFAULT_CIDR_RANGE,
                SubnetConfiguration = new[] {
                    new SubnetConfiguration{
                        Name = Constants.SUBNET_PUBLIC_ID,
                        SubnetType = SubnetType.PUBLIC
                    },
                    new SubnetConfiguration {
                        Name = Constants.SUBNET_PRIVATE_ID,
                        SubnetType = SubnetType.PRIVATE
                    }
                }
            });
            this.VPC = vpc;
            this.VPCID = vpc.VpcId;
            new CfnOutput(this, "VPCId", new CfnOutputProps
            {
                Value = this.VPCID
            });
        }
    }
}

File: python/privatelink-rds/stacks/vpc_stack.py
from aws_cdk import (
    core,
    aws_ec2 as ec2
)

class VpcStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, 
        vpc_cidr, 
        **kwargs
    ) -> None:
        super().__init__(scope, id, **kwargs)

        # SubnetType.ISOLATED used as we don't want Internet traffic possible for this demo
        self.vpc = ec2.Vpc(self, "VPC",
           max_azs = 2,
           cidr = vpc_cidr,
           subnet_configuration = [
               ec2.SubnetConfiguration(
                   subnet_type = ec2.SubnetType.ISOLATED,
                   name = "PrivateIngress",
                   cidr_mask = 28
               ), ec2.SubnetConfiguration(
                   subnet_type = ec2.SubnetType.ISOLATED,
                   name = "DB",
                   cidr_mask = 28
               )
           ],
       )
        core.CfnOutput(self, "Output",
           value = self.vpc.vpc_id)